server/schedule/placement/rule_manager.go

```diff
@@ -305,9 +307,24 @@ func (m *RuleManager) GetRulesForApplyRegion(region *core.RegionInfo) []*Rule {
 }
 
 // FitRegion fits a region to the rules it matches.
-func (m *RuleManager) FitRegion(stores StoreSet, region *core.RegionInfo) *RegionFit {
+func (m *RuleManager) FitRegion(storeSet StoreSet, region *core.RegionInfo) *RegionFit {
+	stores := storeSet.GetStores()
 	rules := m.GetRulesForApplyRegion(region)
-	return FitRegion(stores, region, rules)
+	if m.cache.Check(region, rules, getStoresByRegion(stores, region)) {
+		fit := m.cache.GetCacheRegionFit(region.GetID())
+		if fit != nil {
+			fit.SetCached(true)
+			return fit
+		}
+	}
+	fit := FitRegion(stores, region, rules)
+	if fit.IsSatisfied() && len(region.GetDownPeers()) == 0 && region.GetLeader() != nil {
+		m.cache.SetCache(region, rules, fit, getStoresByRegion(stores, region))
+	} else {
+		m.cache.Invalid(region.GetID())
+	}
+	fit.SetCached(false)
```

Comment:
I'm not quite understand what `Cached` is used for.

Reply:
In future pr, if the `checker` find that the fit is cached, it will skip the current region instead of generating operator.

Reply:
Then I guess we should not set the flag to `true` at L316. Instead, the flag should only be set to `true` by checker when it cannot create any operator to improve the region. Also, I wonder if `checked` is more appropriate than `cached`.